ddlitlab2024/dataset/imports/strategies/bitbots.py

```diff
@@ -0,0 +1,282 @@
+    "/debug/approach_point",
+    "/debug/ball_twist",
+    "/debug/dsd/body_behavior/dsd_current_action",
+    "/debug/dsd/body_behavior/dsd_stack",
+    "/debug/dsd/body_behavior/dsd_tree",
+    "/debug/dsd/hcm/dsd_current_action",
+    "/debug/dsd/hcm/dsd_stack",
+    "/debug/dsd/hcm/dsd_tree",
+    "/debug/dsd/localization/dsd_current_action",
+    "/debug/dsd/localization/dsd_stack",
+    "/debug/dsd/localization/dsd_tree",
+    "/debug/used_ball",
+    "/debug/which_ball_is_used",
+    "/diagnostics",
+    "/diagnostics_agg",
+    "/field/map",
+    "/gamestate",
+    "/goal_pose",
+    "/head_mode",
+    "/motion_odometry",
+    "/pose_with_covariance",
+    "/robot_state",
+    "/robots_relative",
+    "/robots_relative_filtered",
+    "/rosout",
+    "/strategy",
+    "/system_workload",
+    "/team_data",
+    "/tf",
+    "/tf_static",
+    "/time_to_ball",
+]
+
+TOPIC_LIST_RBC24 = [
+    "/DynamixelController/command",
+    "/audio/audio",
+    "/ball_position_relative_filtered",
+    "/balls_relative",
+    "/camera/camera_info",
+    "/camera/image_proc",
+    "/cmd_vel",
+    "/core/power_switch_status",
+    "/debug/dsd/body_behavior/dsd_stack",
+    "/debug/dsd/body_behavior/dsd_tree",
+    "/debug/dsd/hcm/dsd_current_action",
+    "/debug/dsd/hcm/dsd_stack",
+    "/debug/dsd/hcm/dsd_tree",
+    "/debug/dsd/localization/dsd_stack",
+    "/debug/dsd/localization/dsd_tree",
+    "/diagnostics",
+    "/diagnostics_agg",
+    "/field/map",
+    "/joint_states",
+    "/motion_odometry",
+    "/pose_with_covariance",
+    "/robot_state",
+    "/robots_relative",
+    "/robots_relative_filtered",
+    "/rosout",
+    "/strategy",
+    "/system_workload",
+    "/tf",
+    "/tf_static",
+    "/time_to_ball",
+    "/workspace_status",
+]
+
+USED_TOPICS = [
+    "/DynamixelController/command",
+    "/camera/camera_info",
+    "/camera/image_proc",
+    "/gamestate",
+    "/imu/data",
+    "/joint_states",
+    "/tf",
+]
+
+
+class BitBotsImportStrategy(ImportStrategy):
+    def __init__(self, metadata: ImportMetadata):
+        self.metadata = metadata
+
+    def convert_to_model_data(self, file_path: Path) -> ModelData:
+        with self._mcap_reader(file_path) as reader:
+            summary: Summary | None = reader.get_summary()
+
+            if summary is None:
+                logger.error("No summary found in the MCAP file, skipping processing.")
+                return ModelData()
+
+            first_used_msg_time = None
+            model_data = ModelData(recording=self.create_recording(summary, file_path))
+            assert model_data.recording is not None, "Recording is not set"
+
+            self._log_debug_info(summary, model_data.recording)
+
+            for _, channel, message, ros_msg in reader.iter_decoded_messages(topics=USED_TOPICS):
+                first_used_msg_time = first_used_msg_time or message.publish_time
+                relative_timestamp = (message.publish_time - first_used_msg_time) / 1e9
```

Comment:
@Flova Do we want to use the `publish_time` or the time from the `header.stamp` if available?

Reply:
For example camera images are stamped when the trigger is activated. But they are available to the software a fraction of a second later. Using the trigger time could lead to an image being available to the network where it would normally be in transit. This could lead to the network expecting it to be available and performing different during inference where it physically isn't available yet.